cmd/spire-server/cli/run/run.go

```diff
@@ -406,33 +427,36 @@ func NewServerConfig(c *Config, logOptions []log.Option, allowUnknownConfig bool
 		}
 
 		federatesWith := map[spiffeid.TrustDomain]bundleClient.TrustDomainConfig{}
-		for trustDomain, config := range c.Server.Federation.FederatesWith {
-			port := defaultBundleEndpointPort
-			if config.BundleEndpoint.Port != 0 {
-				port = config.BundleEndpoint.Port
-			}
-			if config.BundleEndpoint.UseWebPKI && config.BundleEndpoint.SpiffeID != "" {
-				return nil, errors.New("usage of `bundle_endpoint.spiffe_id` is not allowed when authenticating with Web PKI")
-			}
-
-			var spiffeID spiffeid.ID
-			if config.BundleEndpoint.SpiffeID != "" {
-				spiffeID, err = spiffeid.FromString(config.BundleEndpoint.SpiffeID)
-				if err != nil {
-					return nil, err
-				}
-			}
 
+		for trustDomain, config := range c.Server.Federation.FederatesWith {
 			td, err := spiffeid.TrustDomainFromString(trustDomain)
 			if err != nil {
 				return nil, err
 			}
 
-			federatesWith[td] = bundleClient.TrustDomainConfig{
-				EndpointAddress:  fmt.Sprintf("%s:%d", config.BundleEndpoint.Address, port),
-				EndpointSpiffeID: spiffeID,
-				UseWebPKI:        config.BundleEndpoint.UseWebPKI,
+			var trustDomainConfig *bundleClient.TrustDomainConfig
+			switch {
+			case config.DeprecatedBundleEndpoint != nil:
```

Comment:
Do we first need a case statement like 
```case config.DeprecatedBundleEndpoint != nil && config.BundleEndpointProfile != nil```

we should bail out if both are configured?

The way the code is, it looks like you can have a mix of new and old config for different trust domain ... which is probably fine for now.

Reply:
Both the deprecated configuration section and the new setting should not be allowed at the same time, I'll fix that, thanks.

>The way the code is, it looks like you can have a mix of new and old config for different trust domain ... which is probably fine for now.

Yes, that's intentional.